### PR TITLE
[RISCV] Add more files to .github/CODEOWNERS

### PR DESCRIPTION
**Pull request by topperc**

### CODE REVIEW

**Comment by topperc** [comment]
@llvm/pr-subscribers-riscv

**Comment by jrtc27** [comment]
Or perhaps we should follow the suggestion in https://discourse.llvm.org/t/input-needed-teams-for-pull-request-subscriptions/73116/79? That's effectively what my Herald rule did (though case-insensitive).

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -497,6 +497,7 @@ utils/bazel/llvm-project-overlay/libc/** @llvm/pr-subscribers-libc
 /llvm/**/*RISCV* @llvm/pr-subscribers-riscv
 /clang/**/*RISCV* @llvm/pr-subscribers-riscv
 /clang/include/clang/Basic/riscv_*.td @llvm/pr-subscribers-riscv
+/clang/lib/Headers/riscv_* @llvm/pr-subscribers-riscv
```
Comment (jrtc27) [inline .github/CODEOWNERS:500]: Any reason not to do:

```suggestion
/llvm/**/*RISCV* @llvm/pr-subscribers-riscv
/llvm/**/*riscv* @llvm/pr-subscribers-riscv
/clang/**/*riscv* @llvm/pr-subscribers-riscv
/clang/**/*RISCV* @llvm/pr-subscribers-riscv
/lld/**/*riscv* @llvm/pr-subscribers-riscv
/lld/**/*RISCV* @llvm/pr-subscribers-riscv
```
?

Comment (topperc) [reply .github/CODEOWNERS:500]: I was wondering if lld RISC-V should be it's own team? Seems like a much smaller set of interested people.

Comment (jrtc27) [reply .github/CODEOWNERS:500]: I'd be ok with that

Comment (MaskRay) [reply .github/CODEOWNERS]: See also https://discourse.llvm.org/t/input-needed-teams-for-pull-request-subscriptions/73116/80

There is a difference between “core” backend files and catch-all patterns.

Comment (topperc) [reply .github/CODEOWNERS]: @MaskRay Do you have a suggestion for what "core" for RISC-V should be?

**Comment by asb** [comment]
For me personally, `**/*RISCV ` matches what I want and I'm not sure the volume of e.g. RISC-V specific LLD or LLDB patches is enough to justify splitting it out into separate teams....but it looks like I'm going to be handling labeling of any RISC-V related patches on my email client-side anyway, so my personal preference is even less relevant!

This proposed PR is I think definitely an incremental improvement on the current patterns, so I wouldn't be opposed to merging it and moving discussion on the 'ideal' setup to elsewhere. Marking it as LGTM on that basis.

**Review by asb — APPROVED** [review]

**Pull request merged**